guides/source/developers/extensions/decorators.html.md

```diff
@@ -32,6 +32,34 @@ total by $10 (or whatever your currency is).
 
 [monkey-patch]: https://en.wikipedia.org/wiki/Monkey_patch
 
+## Using ActiveRecord methods in decorators
+
+In order to access ActiveRecord methods, you'll need to define a special
+method.
+
+```ruby
+module MyStore::ProductDecorator
+
+  # This is the place to define custom associations, delegations, scopes and
+  # other ActiveRecord stuff
+  def self.prepended(base)
+    base.has_many :comments, dependent: :destroy
+    base.scope    :sellable, -> { base.where(...).order(...) }
+    base.delegate :something, to: :something
+
+    base.singleton_class.prepend ClassMethods
```

Comment:
I think this line is not really needed, we should have the same result without it.

Reply:
To expand a little bit, in the example gist provided they also have:

```ruby
    # Class methods
    module ClassMethods
      def ransackable_associations
        %w(...)
      end
    end
```

which contains methods injected at a class level but we haven't added that module into the example so that prepend is not needed. I'm fine both with and without adding that extra scenario.

Reply:
I'd rather have less code in the example where possible